### PR TITLE
Use requirements.txt for Gubernator dependencies

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Add IntelliJ IDEA files to gitignore

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

Use requirements.txt for Gubernator dependencies

Deploying Gubernator using a custom runtime on Google App Engine or
deploying it without Google App Engine entirely should be possible. To
enable these efforts, the dependency requirements for the application
should be listed in the standard `requirements.txt` manifest instead of
the Google App Engine-specific YAML.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@spxtr how may I join the Kubernetes organization?

**Comment by spxtr** [comment]
/ok-to-test

I'm not sure where it's documented, but I think if you are a regular contributor it will happen eventually. In the meantime, it's not hassle for us to say `/ok-to-test` :)

**Comment by stevekuznetsov** [comment]
Sounds good!

File: gubernator/app.yaml [hunk 1]
```diff
@@ -18,9 +18,3 @@ handlers:
 - url: .*
   script: main.app
   secure: always
-
-libraries:
```
Comment (rmmh) [inline gubernator/app.yaml]: Leave this in, I still need it for GAE deployments.

**Comment by fejta** [comment]
Would you be interested in converting gubernator to a kubernetes application?

**Comment by stevekuznetsov** [comment]
I've got a WIP to deploy Gubernator as a k8s application right now on our OpenShift CI server -- @fejta I'm happy to contribute the Deployment, Service, etc here if that is something `test-infra` is interested in.

File: gubernator/app.yaml (hunk 1 above)
Comment (stevekuznetsov) [inline gubernator/app.yaml]: Ah, the doc made it seem like GAE would pick up the `requirements.txt`

Comment (stevekuznetsov) [inline gubernator/app.yaml]: @rmmh updated

**Comment by rmmh** [comment]
The build results display should be relatively easy to deploy standalone, but the PR dashboard is probably a lot more difficult, and would likely need to be disabled altogether.

**Comment by stevekuznetsov** [comment]
> but the PR dashboard is probably a lot more difficult, and would likely need to be disabled altogether.

@rmmh what challenges do you anticipate?

**Pull request merged**